R/mod_hierarchical_count_table.R

```diff
@@ -43,135 +217,226 @@ EC <- poc( # nolint
 
-  reduced_group_event_df <- local({
-    reduced_group_event_df <- event_df[, c(subjid_var, hierarchy), drop = FALSE]
-    reduced_group_event_df <- dplyr::left_join(
-      reduced_group_event_df, no_na_pop_df[, c(subjid_var, group_var), drop = FALSE],
-      by = subjid_var
-    )
-  })
-  # event_df may not contain all groups
-  levels(reduced_group_event_df[[group_var]]) <- levels(no_na_pop_df[[group_var]])
-
-  rev_hierarchy <- rev(hierarchy)
-  hierarchy_df_list <- vector(mode = "list", length = length(rev_hierarchy))
-  for (idx in c(seq_along(rev_hierarchy), length(rev_hierarchy) + 1)) {
-    curr_hierarchy <- local({
-      if (idx > length(rev_hierarchy)) character(0) else rev_hierarchy[idx:length(rev_hierarchy)]
-    })
-    empty_hierarchy <- setdiff(hierarchy, curr_hierarchy)
-
-    curr_group_by_cols <- c(curr_hierarchy, group_var)
-    curr_group_select_cols <- c(subjid_var, curr_hierarchy, group_var)
-
-    curr_total_by_cols <- c(curr_hierarchy)
-    curr_total_select_cols <- c(subjid_var, curr_hierarchy)
-
-    # Explicit inclusion of group_var in complete even for totals
-    complete_by <- c(curr_hierarchy, group_var)
-
-    curr_grouped_count_df <- reduced_group_event_df |>
-      dplyr::select(dplyr::all_of(curr_group_select_cols)) |>
-      dplyr::distinct() |>
-      dplyr::group_by(dplyr::across(dplyr::all_of(curr_group_by_cols))) |>
-      dplyr::summarise(
-        "N" = dplyr::n(),
-        subjid = list(.data[[subjid_var]]),
-        perc = {
-          current_group <- dplyr::cur_group()[[group_var]]
-          100 * (.data[["N"]] / n_denominator[[current_group]])
-        }
-      ) |>
-      dplyr::ungroup() |>
-      tidyr::complete(!!!rlang::syms(complete_by), fill = list(N = 0, subjid = list(character(0)), perc = 0))
-
-    curr_total_count_df <- reduced_group_event_df |>
-      dplyr::select(dplyr::all_of(curr_total_select_cols)) |>
-      dplyr::distinct() |>
-      dplyr::group_by(dplyr::across(dplyr::all_of(curr_total_by_cols))) |>
-      dplyr::summarise(
-        "N" = dplyr::n(),
-        subjid = list(.data[[subjid_var]]),
-        perc = {
-          current_group <- total_column_name
-          100 * (.data[["N"]] / n_denominator[[current_group]])
-        }
-      ) |>
-      dplyr::ungroup() |>
-      dplyr::mutate(
-        !!group_var := factor(total_column_name)
-      ) |>
-      tidyr::complete(
-        !!!rlang::syms(complete_by),
-        fill = list(N = 0, subjid = list(character(0)), perc = 0)
-      )
+  if (has_origin_dt) {
+
+    # NOTE!! Partial dates end up as NA!
+    pop_df2[[".origin_dt"]] <- as.Date(pop_df2[[origin_date_var]])
+    if (any(is.na(pop_df2[[".origin_dt"]]) & !is.na(pop_df2[[origin_date_var]])))
+      log_inform("Partial origin dates failed to convert to date format!", level = "inform")
+  }
+
+  if (has_censor_dt) {
+
+    # NOTE!! Partial dates end up as NA!
+    pop_df2[[".censor_dt"]] <- as.Date(pop_df2[[censor_date_var]])
```

Comment:
Is there a possible name collision? If you are using these IDs somewhere else I would include them in ID poc, or store them in variables with meaningful names. Applies to the three `if` statements.